### PR TITLE
executor: remove Next for GrantExec

### PR DESCRIPTION
**Pull request by colinback**
It looks like Next function in GrantExec is an implementation that does nothing.
I just remove the Next function and `go test ./executor` run clean.

BTW,  master branch that I fork has one test failing. Am I the only one that have this FAIL?
```
OOPS: 74 passed, 1 skipped, 1 FAILED
```

unused check has two places that is not related with GrantExec.
```
src/github.com/pingcap/tidb/executor/executor.go:898:2: field asName is unused (U1000)
src/github.com/pingcap/tidb/executor/show.go:61:2: field fetched is unused (U1000)
```

### CODE REVIEW

File: executor/grant.go [hunk 1]
```diff
@@ -52,15 +52,6 @@ type GrantExec struct {
...
-func (e *GrantExec) Next(ctx context.Context) (Row, error) {
-	if e.done {
-		return nil, nil
-	}
-	e.done = true
-	return nil, errors.Trace(e.run(ctx))
-}
-
 // NextChunk implements the Executor NextChunk interface.
 func (e *GrantExec) NextChunk(ctx context.Context, chk *chunk.Chunk) error {
```
Comment (zz-jason) [inline executor/grant.go]: after removing `Next`, `e.run` is only called by `e.NextChunk`, we'd better remove `e.run` and put the codes into `e.NextChunk`

**Comment by zz-jason** [comment]
@colinback could you please provide us the failed UT case ?

File: executor/grant.go (hunk 1 above)
Comment (colinback) [reply executor/grant.go]: Got it. Sorry I have to go out now; I will continue later today or tomorrow.

**Comment by zz-jason** [comment]
to #5985

**Comment by colinback** [comment]
@zz-jason it looks like fails at the 'make dev' step.
First I run on my macbook pro and go version  is 'go1.9.2 darwin/amd64';
Then I test on Ubuntu with go version 'go1.9.4 linux/amd64', but still the same
issue. Following is the my testing level.
```
commit 4a52a6ae580224a62b2c17707e64720af26c88fd
Author: HuaiyuXu <391585975@qq.com>
Date:   Fri Mar 9 21:19:24 2018 +0800

    executor: remove Next for HashJoinExec (#5991)
```

make dev:
```
ok     	github.com/pingcap/tidb/util/testutil  	0.007s 	coverage: 46.5% of statements
?      	github.com/pingcap/tidb/x-server       	[no test files]
go get github.com/kisielk/errcheck
go get github.com/golang/lint/golint
vet
golint
gofmt (simplify)
./util/arena/arena.go
make: *** [check] Error 1
```

```
diff --git a/util/arena/arena.go b/util/arena/arena.go
index a8f8e1ab..94729b19 100644
--- a/util/arena/arena.go
+++ b/util/arena/arena.go
@@ -60,7 +60,7 @@ func NewAllocator(capacity int) *SimpleAllocator {
 // Alloc implements Allocator.AllocBytes interface.
 func (s *SimpleAllocator) Alloc(capacity int) []byte {
        if s.off+capacity < cap(s.arena) {
-               slice := s.arena[s.off : s.off : s.off+capacity]
+               slice := s.arena[s.off:s.off : s.off+capacity]
                s.off += capacity
                return slice
        }
```

Then `go test` fails for TestT:
```
2018/03/10 02:18:29.766 ddl.go:341: [info] close DDL:f02bb6c8-5a30-4d2b-be98-c879d81d4d73
2018/03/10 02:18:29.766 ddl.go:304: [info] stop DDL:f02bb6c8-5a30-4d2b-be98-c879d81d4d73
OOPS: 74 passed, 1 skipped, 1 FAILED
--- FAIL: TestT (98.59s)
FAIL
exit status 1
FAIL   	github.com/pingcap/tidb	98.614s
```

**Comment by zz-jason** [comment]
@colinback we use `go` 1.10 now, so maybe you should update your go version to 1.10 as well

**Pull request merged**